addons/product/models/product.py

```diff
@@ -658,7 +658,10 @@ def _get_filtered_sellers(self, partner_id=False, quantity=0.0, date=None, uom_i
         precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
 
         sellers_filtered = self._prepare_sellers(params)
-        sellers_filtered = sellers_filtered.filtered(lambda s: not s.company_id or s.company_id.id == self.env.company.id)
+        if self.env.company.id in sellers_filtered.company_id.ids:
```

Comment:
I would set in the context the company of the object you want a seller for (for instance a purchase order).
Here check the context, if there is a specific company requested: 
`sellers_filtered.filtered(lambda s: not s.company_id or s.company_id.id == context_company.id)`
if not,
`sellers_filtered.filtered(lambda s: not s.company_id or s.company_id.id == self.env.company.id)`
It can be wrong to take a supplier from `self.env.companies` as this can have nothing related to the company of the document you are working on